### DISCUSSION TITLE
What does "number of graphs" mean in Homogeneous Graph Data?

### DISCUSSION THREAD

**Discussion by Srivathsan-Srinivas** [pyg-team/pytorch_geometric#9438] [Q&A]
I created a custom Homogeneous Graph Dataset:
    `graph_data = Data(x=node_features_tensor, edge_index=edge_index_tensor, y=labels_tensor)`
    
  
When I print the number of graphs, it says 3.

Graph Dataset:
```
Data(x=[40, 92], edge_index=[2, 158], y=[40])
Number of graphs: 3
Number of nodes: 40
Number of edges: 158
Number of node-features: 92
Number of edge-features: 0
Edges are directed: False
Edge Index Shape = torch.Size([2, 158])
Graph has isolated nodes: False
Graph has loops: True
Edge Attributes: 40
```

I am a beginner to PyG. I just learnt how to create a PyG homogeneous dataset from custom input (csv) file. The exact print statement is:
`print(f"Number of graphs: {len(graph_data)}")`

Should I be worried about this? The reason why I am asking is : When I studied GNNs, the tutorials start with Zachary's Karate Club example....and that shows exactly 1 graph. Not sure how..as the dataset is prepared in advance. I am trying to create my own dataset here...and want to ensure I am not making a mistake.

To double-check my graph-data, I converted to network graph and plotted it. It turned out to be a single, connected graph. The two yellow nodes have different label than the rest of the nodes.

![homogeneous_graph](https://github.com/pyg-team/pytorch_geometric/assets/923893/73fe8ac7-688d-47a2-81ee-11d2322e3ca2)

**Comment by rusty1s** [comment]
`len(data)` will return the number of attributes this `Data` object is holding.
